test/ambiguity.jl

```diff
@@ -0,0 +1,57 @@
+# ambiguity.jl
+# tests needed only because of code added to resolve method ambiguities
+
+using LinearAlgebra: Diagonal, UpperTriangular
+using LinearAlgebra: Adjoint, Transpose, Symmetric
+using LinearAlgebra: TransposeAbsVec, AdjointAbsVec
+import LinearAlgebra
+using LinearMapsAA
+using Test: @test, @testset
+
+M = rand(3,3)
+A = LinearMapAA(M)
+
+@testset "Diagonal" begin
+    D = Diagonal(1:3)
+    @test Matrix(A * D) == M * D
+    @test Matrix(D * A) == D * M
+end
+
+@testset "UpperTri" begin
+    U = UpperTriangular(M)
+    @test Matrix(A * U) == M * U
+    @test Matrix(U * A) == U * M
+end
+
+@testset "TransposeVec" begin
+    xt = Transpose(1:3)
```

Comment:
Same here: why not `transpose(1:3)`? `Transpose` is a "constructor", `transpose` is a function, that calls the constructor or does something else, depending on the type.